app/components/container/containerController.js

```diff
@@ -197,6 +197,104 @@ function ($scope, $state, $stateParams, $filter, Container, ContainerCommit, Con
     });
   };
 
+  $scope.recreate = function() {
+    var config = $scope.container.Config;
+    // HostConfig
+    config.HostConfig = $scope.container.HostConfig;
+    // Name
+    config.name = $scope.container.Name.replace(/^\//g, '');
+    // Network
+    var mode = config.HostConfig.NetworkMode;
+    // TODO get ip addrs
+    config.NetworkingConfig = {
+      "EndpointsConfig": {}
+    };
+    config.NetworkingConfig.EndpointsConfig = $scope.container.NetworkSettings.Networks;
+    if (mode.indexOf('container:') !== -1) {
+      delete config.Hostname;
+      delete config.ExposedPorts;
+    }
+    // Set volumes
+    var binds = [];
+    var volumes = {};
+    for (var v in $scope.container.Mounts) {
+      console.log($scope.container.Mounts[v]);
+      var mount = $scope.container.Mounts[v];
+      var volume = {
+        "type": mount.Type,
+        "name": mount.Name || mount.Source,
+        "containerPath": mount.Destination,
+        "readOnly": mount.RW === false
+      };
+
+      var name = mount.Name || mount.Source;
+      var containerPath = mount.Destination;
+      if (name && containerPath) {
+        var bind = name + ':' + containerPath;
+        volumes[containerPath] = {};
+        if (mount.RW === false) {
+          bind += ':ro';
+        }
+        binds.push(bind);
+      }
+    }
+    config.HostConfig.Binds = binds;
+    config.Volumes = volumes;
+
+    Container.remove({id: $scope.container.Id, v: 0, force: true}, function(d) {
```

Comment:
Have to play with users and teams struct, cause object have AccessLevel too, see https://github.com/portainer/portainer/pull/855/commits/6d9c1c99bb641a32d300557f78770d831e4ab8e4